doc/reference/trd-appid.md

```diff
@@ -0,0 +1,396 @@
+  identifier. The application credentials of each Tock binary of the
+  process consist a TBF header containing the public key and a
+  signature of the SHA512 of the application binary made with the
+  corresponding private key.
+
+An application identifier provides an identity for an application
+binary. It allows the Tock kernel to know about the provenance and
+origin of the binary and make access control or security decisions
+based on this information. For example, a kernel may allow only applications whose
+credentials use a particular trusted public key to access restricted
+functionality, but restrict other applications to use a subset of
+available system calls.
+
+Application identifiers are distinct from process
+identifiers; an application identifier is per-application (persists
+across restarts of a Tock binary, for example), while a process
+identifier identifies a particular execution of that binary. At any
+time on a Tock device, each process has a unique process identifier,
+but they can be re-used over time (like POSIX process identifiers).
+
+As the above examples illustrate, application credentials can vary in
+size and content. The credentials that the kernel will accept depends
+on its use case: certain devices will only accept credentials which
+include a particular public key, while others will accept
+many. Furthermore, the internal format of these credentials can vary.
+Finally, the cryptography used in credentials can vary, either due to
+security policies or certification requirements.
+
+4. Credentials in Tock Binary Format Headers
+===============================
+
+To support credentials in Tock binaries, the Tock Binary Format has
+a `TbfHeaderV2Credentials` header. This header is variable length
+and has two fields:
+
+```rust
+pub struct TbfHeaderV2Credentials {
+    format: TbfHeaderV2CredentialsType,
+    data: &[u8],
+}  
+```
+
+The `TbfHeaderV2CredentialsType` defines the format and size of `data`
+field. A `TbfHeaderV2CredentialsType` value MUST have a fixed
+data size and format. Currently supported values are:
+
+```rust
+pub enum TbfHeaderV2CredentialsType {
+    CleartextID = 0,
+    Rsa3072Key = 1,
+    Rsa4096Key = 2,
+    Rsa3072KeyWithID = 3,
+    Rsa4096KeyWithID = 4,
+}
+```
+
+**These are not intended to be final or prescriptive. They are merely some examples
+of what kind of information we might put here. Among other things, the exact format 
+of the data blocks needs to be more precise. -pal**
+
+The `CleartextID` value has a data length of 8 bytes. It contains a 64-bit number in
+big-endian format representing an application identifier.
+
+The `Rsa3072Key` value has a data of length of 768 bytes. It contains a public 3072-bit
+RSA key (384 bytes), followed by a 384-byte ciphertext block, consisting of the SHA512 
+hash of the application binary in this Tock binary, encrypted by the private key 
+of the public key in the header.
```

Comment:
The nomenclature is "signed", not "encrypted". RSA signing and encryption/decryption are not the same thing: https://www.cs.cornell.edu/courses/cs5430/2015sp/notes/rsa_sign_vs_dec.php

Reply:
Good catch.